internal/internal_event_handlers.go

```diff
@@ -933,6 +934,11 @@ func (weh *workflowExecutionEventHandlerImpl) handleLocalActivityMarker(markerDa
 	}
 
 	if la, ok := weh.pendingLaTasks[lamd.ActivityID]; ok {
+		if len(lamd.ActivityType) > 0 && lamd.ActivityType != la.params.ActivityType {
```

Comment:
We could have a separate counter that includes al events including local activity scheduling events and pass that counter down to localActivityResult to ensure determinism.